pkg/pki/certs.go

```diff
@@ -43,6 +47,11 @@ func EnsureKeypair(store MutableKeypair, config certutil.Config, signer *Keypair
 
 			match := true
 
+			if match && time.Until(cert.NotAfter) <= RENEW_THRESHOLD {
```

Comment:
So I _think_ I hadn't intended to persist these keys at all.

One option is just to use a in memory key store; but I do like what you have here.  A twist on this could be to regenerate if the key was issued > 24 hours ago, which is similar to no persistence, but offers some consistency over the short term.

Reply:
(I'm trying to figure out if there's a reason why we do persist these keys, when we don't persist others)

Reply:
Yeah, it’s possible that all this logic should be removed and just regen the cert on every startup. My only worry was entropy depletion since remaking the key would spike that pretty hard during system startup. But just the cert shouldn’t be an issue I think?

Reply:
Good point that this is a cheaper operation because it's only the cert, not the key.

I think this fix is the right one to do immediately; I might tweak the threshold, but I'd rather get this merged and then look at what happens e.g. if I take that threshold effectively to zero.